src/Compilers/CSharp/Test/Emit/PDB/CSharpPDBTestBase.cs

```diff
@@ -17,6 +20,64 @@ namespace Microsoft.CodeAnalysis.CSharp.UnitTests.PDB
 {
     public class CSharpPDBTestBase : CSharpTestBase
     {
+        public static void TestLambdasAndClosures(string input, string methodStartSpanName = "method", string closureSpanName = "closure", string lambdaSpanName = "lambda")
+        {
+            MarkupTestFile.GetSpans(input, out var source, out IDictionary<string, ImmutableArray<TextSpan>> spans);
+
+            var compilation = CreateCompilationWithMscorlib40AndSystemCore(source, options: TestOptions.DebugDll);
+            compilation.GetDiagnostics().Where(d => d.Severity == DiagnosticSeverity.Error).Verify();
```

Comment:
I don't think we should create compilation here. That makes the test helper too specific (what if I need to add references, change options, etc.). It'd be better to take compilation and spans parameters.

Reply:
I was trying to keep the test methods as small as possible, but can certainly add an overload to allow for more customization.

Reply:
Making the test as small as possible is desirable. However, the test helpers need to be flexible, composable and not combine many concepts. Like a product API.

Reply:
We can simplify the test methods like so (other EnC tests already use a similar approach, see [`MarkedSource`](http://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp.Emit.UnitTests/Emit/EditAndContinue/EditAndContinueTestBase.cs,41) method):

```
void Test()
{
   var source = MarkedSource("class C { void Main() <METHOD>{ F(<LAMBDA:0>() => 1}) }");
   var compilation = CreateCompilationWithMscorlib40AndSystemCore(source.Tree, options: TestOptions.DebugDll);
   compilation.VerifyLambdasAndClosures(source.Markers);
}
```

Reply:
This approach allows us to create custom compilations and verify more data then just the PDB lambda and closure entries, if needed.